x/params/store/store.go

```diff
@@ -0,0 +1,177 @@
+package store
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Individual parameter store for each keeper
+type Store struct {
+	cdc  *codec.Codec
+	key  sdk.StoreKey
+	tkey sdk.StoreKey
+
+	space []byte
+}
+
+// NewStore constructs a store with namestore
+func NewStore(cdc *codec.Codec, key sdk.StoreKey, tkey sdk.StoreKey, space string) Store {
+	return Store{
+		cdc:  cdc,
+		key:  key,
+		tkey: tkey,
+
+		space: []byte(space),
+	}
+}
+
+// Returns a KVStore identical with ctx,TransientStore(s.key).Prefix()
+func (s Store) kvStore(ctx sdk.Context) sdk.KVStore {
+	return ctx.KVStore(s.key).Prefix(append(s.space, '/'))
```

Comment:
Why do we append a forward slash?

Reply:
Paramstore spaces / parameter keys should be human readable, so to avoid confusion when there is no separation between the space and the key, we are using a slash as the separator.

Reply:
OK, it costs a bit of storage but we don't need thousands of parameters. :+1: